File: HorseRacing.Contracts/Models/Game/Requests/JoinGameWithBetRequest.cs
using HorseRacing.Contracts.Base.Requests;
using HorseRacing.Contracts.Models.Game.Dtos;

namespace HorseRacing.Contracts.Models.Game.Requests
{
    public class JoinGameWithBetRequest : BaseRequest<JoinGameWithBetRequestDto>
    {
        public JoinGameWithBetRequest() : base() { }
        public JoinGameWithBetRequest(JoinGameWithBetRequestDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/requests/join-game-with-bet/join-game-with-bet-request.ts
import type { BaseRequest } from "~/interfaces/api/contracts/base/requests/base-request";
import type { JoinGameWithBetRequestDto } from "./join-game-with-bet-request-dto";

export class JoinGameWithBetRequest implements BaseRequest<JoinGameWithBetRequestDto> {
    Data: JoinGameWithBetRequestDto;

    constructor(data?: JoinGameWithBetRequestDto) {
        this.Data = data || { UserId: '', GameId: '', BetAmount: 0, BetSuit: 0 };
    }
}
